### DISCUSSION TITLE
Is there encryption at rest available or any solution alike?

### DISCUSSION THREAD

**Discussion by dougg0k** [electric-sql/electric#719] [Questions] [answered]
Title

**Comment by thruflo** [comment, marked as answer]
Not yet. There’s a proposal for hooks https://github.com/electric-sql/electric/discussions/565 which would support encryption of non primary key values of the local device. So plaintext in SQLite, encrypted once it goes on the Satellite protocol and in Postgres. However, this is complex to work with multi-user sharing, so would only be useful for some use cases.

**Comment by rikur** [comment]
https://electric-sql.com/docs/integrations/drivers/mobile/react-native
https://ospfranco.notion.site/Encryption-58aa9d33736f43e9a53a4f6f9116c36a

**Reply by thruflo** [reply]
Yup, you can encrypt the client side database. One challenge is that we also persist data in the server side database if you sync it. So we need a solution to also encrypt over the wire and through Postgres as well as the client storage.

That said, in many cases encrypting on the client is very useful and can provide security against the local data being accessed by other apps or systems on the client.

**Comment by KyleAMathews** [comment]
👋 we've been working the last month on a rebuild of the Electric server over at a temporary repo https://github.com/electric-sql/electric-next/

You can read more about why we made the decision at https://next.electric-sql.com/about

We're really excited about all the new possibilities the new server brings and we hope you'll check it out soon and give us your feedback.

We're now moving the temporary repo back here. As part of that migration we're closing all the old issues, PRs, and discussions. We really appreciate you taking the time to start this discussion! If it's still relevant in the new system, please reopen.
